caffe2/CMakeLists.txt

```diff
@@ -1257,10 +1267,13 @@ install(TARGETS torch torch_library EXPORT Caffe2Targets DESTINATION "${TORCH_IN
 
 target_link_libraries(torch PUBLIC torch_cpu_library)
 if(BUILD_SPLIT_CUDA)
-  target_link_libraries(torch PUBLIC torch_cuda_cu_library)
-  target_link_libraries(torch PUBLIC torch_cuda_cpp_library)
-elseif(USE_CUDA)
-  target_link_libraries(torch PUBLIC torch_cuda_library)
+  target_link_libraries(torch_cuda PUBLIC torch_cuda_cu_library)
+  target_link_libraries(torch_cuda PUBLIC torch_cuda_cpp_library)
+endif()
+
+if(USE_CUDA)
+  # we want this to be run for BUILD_SPLIT_CUDA as well
+  target_link_libraries(torch PUBLIC torch_cuda_library) # should this be torch_cuda?
```

Comment:
I felt like this should be ordered like:
```
IF(USE_CUDA)
  target_link_libraries(torch PUBLIC torch_cuda_library) 
  IF(BUILD_SPLIT_CUDA)
    target_link_libraries(torch_cuda PUBLIC torch_cuda_cu_library)
    ...
  ENDIF() # BUILD_SPLIT_CUDA
ENDIF() # USE_CUDA
```

Reply:
this way you dont need those comments since its self-explanatory

Reply:
+1, nested `if` would be better than unrolled if->elseif->endif chain